test/image/mocks/line_scatter.json

```diff
@@ -30,7 +30,8 @@
         9
       ],
       "mode": "lines",
-      "type": "scatter"
+      "type": "scatter",
+      "opacity": 0.5
```

Comment:
First, see

![image](https://cloud.githubusercontent.com/assets/6675409/21781824/19f67712-d67f-11e6-9b16-5f574b3f6afc.png)

> trace opacity should get set on traceGroup

that would imply that the trace name (rendered in the `<text class="legendtext">` node) would be semi-transparent whenever `trace.opacity < 1`. 

So, in https://github.com/plotly/plotly.js/pull/1204/commits/d0097980ae8786d4d77348de585a8946771bcfc9 a new `<g>` is added grouping the fill, lines and symbols `<g>` and applying `trace.opacity` once.